### PR TITLE
Remove trailing whitespace

### PR DESCRIPTION
**Pull request by ktbarrett**
Remove trailing whitespace in all files and adds trailing whitespace checks to Travis.

### CODE REVIEW

**Comment by eric-wieser** [comment]
@imphil, ok with this in principle? (Obviously not for 1.3)

If so, I'll go through the whitespace-sensitive files and check that nothing is relying on trailing whitespace.

File: .travis.yml [hunk 1]
```diff
@@ -14,6 +14,10 @@ services:
 
 jobs:
   include:
+    - stage: PythonLint
+      os: linux
+      script: "[[ -z \"$(git ls-files | xargs egrep -l '\s+$')\" ]]"
```
Comment (ktbarrett) [inline .travis.yml]: Still not too sure about this, I tried running this without the whitespace changes to see if it failed, but I guess it saw there were no changes and didn't run Travis.

Comment (eric-wieser) [reply .travis.yml]: I think travis might have failed to parse the yaml

Comment (eric-wieser) [reply .travis.yml]: This run is from an older version: https://travis-ci.org/cocotb/cocotb/jobs/632467451#L183

Comment (eric-wieser) [inline .travis.yml]: I think the backslash needs escaping for yaml:
```suggestion
      script: "[[ -z \"$(git ls-files | xargs egrep -l '\\s+$')\" ]]"
```

Comment (ktbarrett) [reply .travis.yml]: Yeah, that's all it needed. Travis failed with master sources and passed with fixed sources.

**Comment by imphil** [comment]
Yes, that's a great change and enforcing this in CI is the right move. Thanks!

**Comment by eric-wieser** [comment]
Thanks @ktbarrett. This is probably going to create conflicts in some open PRs, but hopefully they'll be easy to resolve

**Pull request merged**